### PR TITLE
add supports for cmd: terraform add

### LINKED ISSUE #208: Add supports for `terraform add` command
The v1.1.0 of `terraform` introduces a new subcommand: `terrraform add`.

It would be very helpful to have it available in `terraform-exec`, which can be used for implementing some "bulk" import feature.

### PR DESCRIPTION
**Pull request by magodo**
Fix #208 

Though the official release of the v1.1.0 is not shipped yet, while the alpha version is available. This means general users can use the `terraform add` feature already via the alpha CLI tool. As the `tfinstall` is able to install the alpha release, it means the `tfexec` users should also be able to use the alpha features (i.e. the `add`).

For the reason above, I've added a `Latest_v1_1_alpha` version for the test.

### CODE REVIEW

File: tfexec/internal/testutil/tfcache.go
Comment (radeksimko) [inline tfexec/internal/testutil/tfcache.go:18]:
```suggestion
	Latest_v1_1 = "1.1.0-alpha20210811"
```

File: tfexec/add.go
Comment (radeksimko) [inline tfexec/add.go:45]:
```suggestion
func (tf *Terraform) Add(ctx context.Context, address string, opts ...AddOption) (string, error) {
```

The `add` command writes configuration to stdout by default and so I think that the interface of this method should reflect that default behaviour and probably just entirely omit support for the `out` flag.

IMHO in contexts where `terraform-exec` is used users/developers can still process the string and write it to a file if they choose to.

File: tfexec/internal/testutil/tfcache.go
Comment (radeksimko) [reply tfexec/internal/testutil/tfcache.go:18]: Technically this is an internal constant anyway, so it probably doesn't matter if we later rename it but I expect that we'd more likely be testing against any latest v1.1 once a stable version is out rather than keep testing against alpha versions.

File: tfexec/add.go
Comment (radeksimko) [inline tfexec/add.go:44]:
```suggestion
// Add represents the `terraform add` subcommand (added in 1.1.0).
//
// Note that this function signature and behaviour is subject
// to breaking changes including removal of that function
// until final 1.1.0 Terraform version (with this command) is released.
```

**Review by radeksimko — CHANGES_REQUESTED** [review]
Thank you for the PR @magodo 

I wouldn't say it's necessarily a rule that any `tfinstall`-installable version has to be supported in `tfexec` but as long as it's correctly version-gated then I think it's fine! 😄 

The `add` command may still undergo some changes before getting out of alpha, but it's also equally likely that it will just land as is. With that in mind I also proposed a minor change in the comment, so that folks are more aware of potential BCs. That said they probably will be aware anyway as they have to opt into installing alpha version 🤷🏻‍♂️ 

The PR looks pretty good aside from my in-line comment about the `out` flag and the signature. Once that is addressed I'd be happy to merge it.

File: tfexec/internal/testutil/tfcache.go
Comment (magodo) [reply tfexec/internal/testutil/tfcache.go:18]: This makes sense, thank you!

File: tfexec/add.go
Comment (magodo) [reply tfexec/add.go:45]: Indeed, this is totally a mistake. Thank you for pointing this out :+1:

**Comment by magodo** [comment]
@radeksimko Thank you for the quick feedback! I've resolved the comments above, please take another look :)

File: tfexec/add.go
Comment (radeksimko) [inline tfexec/add.go:56]: Nitpick: Given that we're returning `string` and never deal with raw bytes we may as well make this [`strings.Builder`](https://pkg.go.dev/strings#Builder)
```suggestion
	var outBuf strings.Builder
```

This admittedly applies to most current `bytes` usage in other places, but that can be addressed in a separate PR.

**Review by radeksimko — COMMENTED** [review]
Looking good, I'm happy to merge once that last nitpick is addressed 😃

**Review by radeksimko — APPROVED** [review]

**Pull request merged**
